src/Entity/MetadataDisplayEntity.php

```diff
@@ -473,95 +507,57 @@ public function getTwigVariablesUsed() {
    * @param \Twig\Node\Node $nodes
    *   A Twig Module Nodes object.
    *
-   * @param array $all_variables
-   *   An array to track the variables during recursion to return the accumulated line numbers.
-   *
-   * @param array $set_var
-   *   A string to track variables that reference JSON keys.
-   *
-   * @param array $set_source
-   *   An string to track the referenced JSON key for the above.
-   *
    * @return array
    *   A list of used $variables by this template.
    */
-  private function getTwigVariableNames(ModuleNode|Node|BodyNode $nodes, array $all_variables, string $set_var = '', string $set_source = ''): array {
+  private function getTwigVariableNames(ModuleNode|Node|BodyNode $nodes): array {
     $variables = [];
     foreach ($nodes as $node) {
       $lineno = [$node->getTemplateLine()];
-      $variable_key = '';
-      $parent_path = '';
-      if ($node instanceof NameExpression && !$nodes instanceof NotUnary) {
-        if (!$node->hasAttribute('name')) {
-          continue;
-        }
-        $variable_key = $node->getAttribute('name');
-        if ($variable_key == '_key') {
+      $variable_key = $node->hasAttribute('name')
+                      && $node->getAttribute('name') == 'data'
+        ? $node->getAttribute('name')
+        : '';
+      if ($node->hasAttribute('always_defined')) {
```

Comment:
There is no need to parse anything at all (you can return) if $variable_key is empty (also better approach is to make it NULL instead of ''. That is cheaper and more consistent when checking if.